container/spi/src/main/java/org/jboss/arquillian/container/spi/client/deployment/DeploymentScenario.java

```diff
@@ -32,7 +34,7 @@ public class DeploymentScenario {
     private final List<Deployment> deployments;
 
     public DeploymentScenario() {
-        this.deployments = new ArrayList<Deployment>();
+        this.deployments = new CopyOnWriteArrayList<>();
```

Comment:
More is likely needed here to really be thread-safe. Especially in the `findDefaultDeployment` method.